src/libMetrics/Metrics.cpp

```diff
@@ -39,31 +39,41 @@
 #include "opentelemetry/sdk/metrics/metric_reader.h"
 #include "opentelemetry/sdk/resource/resource.h"
 
-#include "common/Constants.h"
 #include "libUtils/Logger.h"
 
 namespace metrics_sdk = opentelemetry::sdk::metrics;
 namespace metrics_exporter = opentelemetry::exporter::metrics;
 namespace metrics_api = opentelemetry::metrics;
 namespace otlp_exporter = opentelemetry::exporter::otlp;
 
-Metrics::Metrics() { zil::metrics::Filter::GetInstance().init(); }
+Metrics::Metrics() {
+  // c-tor
+  zil::metrics::Filter::GetInstance().init();
+}
 
-void Metrics::Init() {
-  std::string cmp(METRIC_ZILLIQA_PROVIDER);
+void Metrics::Initialize(std::string_view identity /*= {}*/,
+                         std::string provider /* = METRIC_ZILLIQA_PROVIDER*/) {
+  static std::array<std::string_view, 4> allowedIdentities = {
+      "seedprv", "seedpub", "dsguard", "lookup"};
```

Comment:
Use a singleton rather than duplicating the list?

Reply:
Moved all this to a separate aux function so it's not repeated. (regardless, I hate singletons and don't really use them).